Assets/Scripts/ScoreController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using TMPro;
+using UnityEngine;
+
+public class ScoreController : Singleton<ScoreController>
+{
+    public TextMeshProUGUI txt;
+
+    public int CurrentScore { get; private set; } = 0;
+
+    private void Start()
+    {
+        UpdateUI();
+    }
+
+    private void Update()
+    {
+        UpdateUI();
```

Comment:
No need to update score every frame, if we only change the score through a function